narwhals/functions.py

```diff
@@ -936,3 +936,92 @@ def get_level(
             - 'interchange': only metadata operations are supported (`df.schema`)
     """
     return obj._level
+
+
+def scan_csv(
+    source: str,
+    *,
+    native_namespace: ModuleType,
+) -> LazyFrame[Any]:
+    """Lazily read from a CSV file.
+
+    This allows the query optimizer to push down predicates and projections
+    to the scan level, thereby potentially reducing memory overhead.
+    For the libraries that do not support lazy dataframes, the function reads
+    a csv file eagerly and then converts the resulting dataframe to a lazyframe.
+
+    Arguments:
+        source: Path to a file.
+        native_namespace: The native library to use for DataFrame creation.
+
+    Returns:
+        LazyFrame.
+
+    Examples:
+        >>> import dask.dataframe as dd
+        >>> import polars as pl
+        >>> import pyarrow as pa
+        >>> import narwhals as nw
+        >>> from narwhals.typing import IntoFrame
+        >>> from types import ModuleType
+
+        Let's create an agnostic function that lazily reads a csv file with a specified native namespace:
+
+        >>> def agnostic_scan_csv(native_namespace: ModuleType) -> IntoFrame:
+        ...     return (
+        ...         nw.scan_csv("file.csv", native_namespace=native_namespace)
+        ...         .to_native()
+        ...         .collect()
```

Comment:
`collect` isn't guaranteed to be present on a native frame, can we keep it out of `agnostic_scan_csv`? you can just put `collect` as the end of the Polars example, and `compute` at the end of the dask example

either that, or put `collect` before `to_native`